packages/styled/types/tests.tsx

```diff
@@ -217,3 +217,69 @@ const Input5 = styled.input`
   // $ExpectError
   ;<StyledCompWithoutAs as={Section} />
 }
+
+{
+  // Props forwarding through StyledOptions and FilteringStyledOptions
+
+  const fc: React.FC<{ foo: string }> = props => <div {...props} />
+
+  styled(fc, { shouldForwardProp: (prop: 'foo') => true })({})
+
+  styled(fc, { shouldForwardProp: (prop: string) => true })({})
+
+  // $ExpectError
+  styled(fc, { shouldForwardProp: (prop: 'bar') => true })({})
+
+  const shouldForwardProp1 = (prop: 'foo') => true
+  styled(fc, { shouldForwardProp: shouldForwardProp1 })({})
```

Comment:
The same doubt about this one as [here](https://github.com/emotion-js/emotion/pull/2333/files#r795040383)